tuple_test.go

```diff
@@ -89,6 +89,45 @@ func TestTuple_NullTuple(t *testing.T) {
 
 }
 
+func TestTuple_TupleNotSet(t *testing.T) {
+	session := createSession(t)
+	defer session.Close()
+	if session.cfg.ProtoVersion < protoVersion3 {
+		t.Skip("tuple types are only available of proto>=3")
+	}
+
+	err := createTable(session, `CREATE TABLE gocql_test.tuple_not_set_test(
+		id int,
+		coord frozen<tuple<int, int>>,
+
+		primary key(id))`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	const id = 1
+
+	err = session.Query("INSERT INTO tuple_not_set_test(id) VALUES(?)", id).Exec()
```

Comment:
In the test we should have 2 rows, one with data the other without.
After reading row 1 x and y should be set accordingly.
After reading row 2 x and y should be set to 0.

Reply:
Good point @mmatczuk we have to ensure they are reset. I think they should be but a test for it makes sense.

Reply:
It didn't reset it.... I was hoping the delegated code would handle this case but apparently not @mmatczuk .

Reply:
Nvm I messed up the evaluation order of the two rows. It seems to work fine.